tools/YamlSplitter/Program.cs

```diff
@@ -0,0 +1,209 @@
+            foreach (var schemaFile in Directory.EnumerateFiles(schemaFolderPath, "*.schema.json"))
+            {
+                using (var sr = new StreamReader(schemaFile))
+                {
+                    var schema = DocumentSchema.Load(sr, schemaFile.Remove(schemaFile.Length - ".schema.json".Length));
+                    schemas.Add(schema.Title, schema);
+                }
+            }
+
+            return schemas;
+        }
+
+        public static void ProcessFilePair(string ymlInputFile, string ymlOutputFile, string mdFile, Dictionary<string, DocumentSchema> schemas)
+        {
+            var yamlStream = new YamlStream();
+            using (var sr = new StreamReader(ymlInputFile))
+            {
+                yamlStream.Load(sr);
+            }
+            if (yamlStream.Documents.Count != 1) throw new NotSupportedException("Does not support mutiple YAML documents");
+
+            var mime = YamlMime.ReadMime(ymlInputFile);
+            if (!mime.StartsWith(YamlMime.YamlMimePrefix))
+            {
+                Console.WriteLine("Invalid Yaml Mime found in " + ymlInputFile + ", should start with " + YamlMime.YamlMimePrefix);
+                return;
+            }
+            var schemaName = mime.Substring(YamlMime.YamlMimePrefix.Length);
+            if (!schemas.ContainsKey(schemaName))
+            {
+                Console.WriteLine("Schema {0} not found", schemaName);
+                return;
+            }
+            var schema = schemas[schemaName];
+
+            var mdFragments = FragmentModelHelper.LoadMarkdownFragment(mdFile);
+
+            Traverse(yamlStream.Documents[0].RootNode, mdFragments, schema);
+
+            var validFragments = mdFragments.Values.Where(v => v.Properties?.Count > 0).ToList();
+
+            if (validFragments.Count > 0)
+            {
+                StringBuilder sb = new StringBuilder();
+                foreach (var fragment in validFragments)
+                {
+                    sb.AppendLine(fragment.ToString());
+                }
+                var mdFolder = Path.GetDirectoryName(mdFile);
+                if (!Directory.Exists(mdFolder))
+                {
+                    Directory.CreateDirectory(mdFolder);
+                }
+                File.WriteAllText(mdFile, sb.ToString());
+
+                var ymlFolder = Path.GetDirectoryName(ymlOutputFile);
+                if (!Directory.Exists(ymlFolder))
+                {
+                    Directory.CreateDirectory(ymlFolder);
+                }
+                YamlUtility.Serialize(ymlOutputFile, yamlStream.Documents[0].RootNode, mime);
+            }
+        }
+
+        private static void Traverse(YamlNode node, Dictionary<string, MarkdownFragment> fragments, BaseSchema schema, string parentOPath = "", string uid = "")
+        {
+            var oPathPrefix = string.IsNullOrEmpty(parentOPath) ? "" : (parentOPath + "/");
```

Comment:
>"" [](start = 66, length = 2)

use String.Empty?

Reply:
for all practical purposes they are the same, right? All "" instances will be treated as one by compiler.

Reply:
Yes, you're right. I think "" will be treated as different objects before...